explorer/src/components/instruction/token/TokenDetailsCard.tsx

```diff
@@ -96,15 +96,38 @@ function TokenInstruction(props: InfoProps) {
   }, [fetchAccountInfo, mintAddress]); // eslint-disable-line react-hooks/exhaustive-deps
 
   const decimals = mintInfo?.decimals;
-  const attributes = [];
-  for (let key in props.info) {
-    const value = props.info[key];
+  const attributes: JSX.Element[] = [];
+
+  buildAttributes(attributes, props.info, decimals);
+
+  return (
+    <InstructionCard
+      ix={props.ix}
+      index={props.index}
+      result={props.result}
+      title={props.title}
+    >
+      {attributes}
+    </InstructionCard>
+  );
+}
+
+function buildAttributes(
+  attributes: JSX.Element[],
+  info: any,
+  decimals: number | undefined
+) {
+  for (let key in info) {
+    const value = info[key];
     if (value === undefined) continue;
 
     let tag;
     let labelSuffix = "";
     if (value instanceof PublicKey) {
       tag = <Address pubkey={value} alignRight link />;
+    } else if (key === "tokenAmount") {
+      buildAttributes(attributes, info.tokenAmount, decimals);
```

Comment:
I don't think it's necessary to flatten the full token amount object like this:

<img width="1151" alt="Screen Shot 2020-09-23 at 11 36 09 AM" src="https://user-images.githubusercontent.com/1076145/93964030-2c16d300-fd91-11ea-8b3d-709e205dc8d0.png">

How about we stick to displaying the ui amount nicely and skip the other fields?